From 944dfa31d45b9ee101932dfa029ec3452a83cb30 Mon Sep 17 00:00:00 2001
Message-Id: <944dfa31d45b9ee101932dfa029ec3452a83cb30.1432858002.git.chang-joon.lee@intel.com>
From: "Kumar, Kiran S" <kiran.s.kumar@intel.com>
Date: Thu, 14 May 2015 11:39:13 +0530
Subject: [PATCH 1/7] SQUASHME! [VPG]: drm/i915: To fix KW scan issue

config and bpp variables are not used. Hence no need to assign
the values for these variables.

SQUASHME: This patch needs to me squashed with
	commit d276c60fd9d834db12a7779eabf2d89287b65c27
	Author: Arun R Murthy <arun.r.murthy@intel.com>
	Date:   Wed Mar 11 09:56:20 2015 +0530
	FOR_UPSTREAM [VPG]: drm/i915: check if pixel format is supported

Issue: GMINL-9366
Change-Id: I8686961482c0334748d556b03906bbf01e55a62b
Signed-off-by: Kumar, Kiran S <kiran.s.kumar@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 0cbdcc1..d3f6e83 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -10776,17 +10776,12 @@ out_hang:
 	return ret;
 }
 
-static int get_format_config(u32 drm_format, u32 *config, u8 *bpp)
+static int get_format_config(u32 drm_format)
 {
 	int i;
-	for (i = 0; i < ARRAY_SIZE(format_mapping); i++) {
-		if (format_mapping[i].drm_format == drm_format) {
-			*config = format_mapping[i].hw_config;
-			*bpp = format_mapping[i].bpp;
+	for (i = 0; i < ARRAY_SIZE(format_mapping); i++)
+		if (format_mapping[i].drm_format == drm_format)
 			return 0;
-		}
-	}
-
 	return -EINVAL;
 }
 
@@ -10813,9 +10808,7 @@ static int intel_crtc_set_pixel_format(struct drm_crtc *crtc,
 		return hsw_set_pixelformat(crtc, fb->pixel_format);
 	} else if (IS_VALLEYVIEW(dev)) {
 		/* MMIO flip handles this for VLV */
-		u32 *config;
-		u8 *bpp;
-		return get_format_config(fb->pixel_format, config, bpp);
+		return get_format_config(fb->pixel_format);
 	} else {
 		DRM_ERROR("Pixel format change not allowed.\n");
 		return -EINVAL;
-- 
1.9.1

